multicluster/controllers/multicluster/member/gateway_controller.go

```diff
@@ -185,19 +194,43 @@ func (r *GatewayReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *GatewayReconciler) getClusterInfo(gateway *mcsv1alpha1.Gateway) *mcsv1alpha1.ClusterInfo {
-	clusterInfo := &mcsv1alpha1.ClusterInfo{
+func (r *GatewayReconciler) clusterSetMapFunc(a client.Object) []reconcile.Request {
+	clientSet := &mcv1alpha2.ClusterSet{}
+	requests := []reconcile.Request{}
+	if a.GetNamespace() != r.namespace {
+		return requests
+	}
+	ctx := context.TODO()
+	err := r.Client.Get(ctx, types.NamespacedName{Namespace: a.GetNamespace(), Name: a.GetName()}, clientSet)
```

Comment:
Besides ClusterSet creation, do we have ClusterSet update too? We should avoid recreation at ClusterSet update (as the processing can be expensive esp. for labelIdentities).

Reply:
Yes, I refined the codes and filtered the update event with `createAndDeleteEventPredicate`.